state/protocol/state.go

```diff
@@ -60,3 +62,14 @@ type MutableState interface {
 	// It modifies the persistent immutable protocol state accordingly.
 	MarkValid(blockID flow.Identifier) error
 }
+
+// IsNodeStakedAtBlockID returns true if the `identifier` is staked at `blockID` according to `state`.
+func IsNodeStakedAtBlockID(state State, blockID flow.Identifier, identifier flow.Identifier) (bool, error) {
+	identity, err := state.AtBlockID(blockID).Identity(identifier)
+	if err != nil {
+		return false, fmt.Errorf("could not retrieve identity for identifier %v at block id snapshot %v: %w)", identifier, blockID, err)
+	}
```

Comment:
```suggestion
	identity, err := state.AtBlockID(blockID).Identity(identifier)
	if protocol.IsIdentityNotFound(err) {
		return false, nil
	}
	if err != nil {
		return false, fmt.Errorf("could not retrieve identity for identifier %v at block id snapshot %v: %w)", identifier, blockID, err)
	}
```

We can also handle the case where the node isn't in the identity table at all, but the block exists -- we're definitely not staked in this case.